regular_expressions_groups/regular_expressions_groups.md

```diff
@@ -0,0 +1,280 @@
+
+You might wonder what would happen if we added a third mouse type, like a CD34+ humanized mouse model with the code "CD34" in its identifier.  Well, we can string together as many or statements as we need.  We can indicate WT or PBMC or CD34 by using `WT|PBMC|CD34`.  This indicates that any of the three values separated by the pipes are valid at that location in a pattern.
+
+</div>
+
+## Non-Capturing Groups
+
+So far, we have used the vertical pipe (`|`) and a capturing group to describe the pattern of our mouse identifiers:  `(WT|PBMC)\-\d{5,6}`.
+
+But what if you don't need to capture the mouse type?  You might just want to make sure that the identifier is valid and matches the pattern correctly, without extracting any text from the identifier.
+
+If that's the case, you can indicate that it's a **non-capturing group** by adding `?:` after the opening parenthesis, like this:
+
+`(?:WT|PBMC)\-\d{5,6}`
+
+[This will also work](https://regex101.com/r/ko7ySn/1) as a matching regular expression, but won't capture the group in question.
+
+This may seem unnecessary.  Why go to the trouble of creating a non-capturing group?  You could simply use a capturing group but not pay any attention to what you capture.  That's true, but consider that in a long string, you might actually want to capture some data and not capture other data, and getting an extra captured group could be confusing or annoying.  It's useful to be able to group things and **not** capture them.
+
+
+### Quiz: Non-Capturing Groups and the Logical "Or"
+
+You're looking through lab notes written by a colleague who uses codes and abbreviations liberally.  
+
+This investigator is known to write their lowercase initials ("sjp") and a single space followed by either "help" (always in lowercase) or "911" in notes to highlight important potential pitfalls.  
+
+Using the logical "or" and using a non-capturing group (remember: the syntax is `(?:)`), write a regular expression that would allow you to capture any of the "sjp help" or "sjp 911" references.
+
+Put that regular expression in the box below:
+
+[[see answer box below for two alternatives]]
+<script>
+  let input = "@input";
+  input == "sjp (?:help|911)" || input == "sjp (?:911|help)" ;
+</script>
+***
+<div class = "answer">
+
+You'll begin by posing your two alternatives, either `help|911` or `911|help`, the order doesn't matter.  Then, you wrap that in the non-capturing group to give `(?:help|911)` or `(?:911|help)`.  But don't forget your collague's initials and the space!  The final regex should read `sjp (?:help|911)` or `sjp (?:911|help)`.
+
+</div>
+***
+
+## Additional Resources
+
+Groups:
+
+* [Capture Group Numbering & Naming: The Gory Details](https://www.rexegg.com/regex-capture.html) goes into detail about how to add numbering to capture groups, which can be helpful in advanced search and replace functions.  This goes beyond the scope we were able to cover in this module.
```

Comment:
Maybe move the Groups section to below the cheat sheets, it is a great resource but definitely a little intimidating!

Reply:
I moved the Groups section down in a new commit @drelliche!